autopush/endpoint.py

```diff
@@ -449,32 +507,7 @@ def _router_completed(self, response, uaid_data):
                                                            uaid_data))
             return d
         else:
-            self.set_status(response.status_code)
-            self.write(response.response_body)
-            for name, val in response.headers.items():
-                self.set_header(name, val)
-            self.finish()
-
-    #############################################################
-    #                    Utility Methods
-    #############################################################
-    def _db_error_handling(self, d):
-        """Tack on the common error handling for a dynamodb request and
-        uncaught exceptions"""
-        d.addErrback(self._overload_err)
-        d.addErrback(self._response_err)
-        return d
-
-    #############################################################
-    #                    Error Callbacks
-    #############################################################
-    def _token_err(self, fail):
-        """errBack for token decryption fail"""
-        fail.trap(InvalidToken)
-        self.set_status(401)
-        log.msg("Invalid token", **self._client_info())
-        self.write("Invalid token")
-        self.finish()
+            response.respond(self)
```

Comment:
I'd prefer to avoid extra entangling of the handler, and make the RouterResponse handling matching the RouterException, ie, a separate method on the handler that takes it and fills it in. The RouterException handling method could be updated so that the logic to dump the contents is the same.